clang/lib/StaticAnalyzer/Checkers/InvalidatedIteratorChecker.cpp

```diff
@@ -31,14 +31,14 @@ class InvalidatedIteratorChecker
                    check::PreStmt<ArraySubscriptExpr>,
                    check::PreStmt<MemberExpr>> {
 
-  std::unique_ptr<BugType> InvalidatedBugType;
+  const BugType InvalidatedBugType{this, "Iterator invalidated",
+                                   "Misuse of STL APIs"};
 
   void verifyAccess(CheckerContext &C, const SVal &Val) const;
-  void reportBug(const StringRef &Message, const SVal &Val,
-                 CheckerContext &C, ExplodedNode *ErrNode) const;
-public:
-  InvalidatedIteratorChecker();
+  void reportBug(const StringRef &Message, const SVal &Val, CheckerContext &C,
```

Comment:
Since we are changing this code, I guess we do not want to take `StringRef`s as const references.

Reply:
Same for the `SVal`. Done.

Reply:
Followup PR for the analyzer-wide refactor: #76688